docs/man-pages/condor_gpu_discovery.rst

```diff
@@ -15,18 +15,31 @@ Description
 -----------
 
 *condor_gpu_discovery* outputs ClassAd attributes corresponding to a
-host's GPU capabilities. It can presently report CUDA and OpenCL
+host's GPU capabilities. It can presently report CUDA, HIP and OpenCL
 devices; which type(s) of device(s) it reports is determined by which
 libraries, if any, it can find when it runs; this reflects what GPU jobs
 will find on that host when they run. (Note that some HTCondor
 configuration settings may cause the environment to differ between jobs
 and the HTCondor daemons in ways that change library discovery.)
 
+This tool is not available for MAC OS platforms.
+
+The ``-cuda``, ``-hip`` and ``-opencl`` arguments control which libraries
+are used for discovering GPUs. If any of these libraries report GPUs,
+then detection will stop at that library unless more then one
+of the above command line arguments is used. If none if these arguments is used,
+*condor_gpu_discovery* will first try to detect GPU devices using
+the NVidia driver library, then the HIP 6 library, and
+finally the OpenCL library. Because OpenCL devices do not have unique
+identifiers, If the ``-opencl`` argument is
+used along with either ``-hip`` or ``-cuda``, OpenCL devices will
+be reported only if they are the only type of device reported.
+
 If ``CUDA_VISIBLE_DEVICES`` or ``GPU_DEVICE_ORDINAL`` is set in the
 environment when *condor_gpu_discovery* is run, it will report only
-devices present in the those lists.
-
-This tool is not available for MAC OS platforms.
+NVidia devices present in the those lists.  If ``HIP_VISIBLE_DEVICES``
+or ``ROCR_VISIBLE_DEVICES`` is set in the environment, it will report
+only HIP devices present on those lists.
```

Comment:
Which list wins?  Or is it the intersection?

Reply:
union